### DISCUSSION TITLE
Custom loss function in AutoMLForecast.fit Hyperparameter tuning

### DISCUSSION THREAD

**Discussion by ArnodeVries** [Nixtla/mlforecast#430] [Q&A] [answered]
I'm new to implementing time series forecasting using the MLForecast library and ran into a problem specifying the loss function for hyperparameter tuning. 

I'm using the AutoMLForecast.fit method to find the best parameters for multiple models. One of the parameters I want to tweak is the loss function. In the [documentation](https://nixtlaverse.nixtla.io/mlforecast/auto.html#automlforecast-fit) I see that this parameter requires some kind of function that takes the validation and the trainings data as input. I can't figure out how to set this up properly to use for example MAE. 

I've tried using the UtilsForecast (from utilsforecast.losses import mae), but that doesn't seem to work.

Anyone who has an idea which utils I can use or how to write them myself?

**Comment by jmoralez** [comment, marked as answer]
Hey. You can use MAE with the following:
```python
from utilsforecast.losses import mae

def loss_fn(df, train_df):
    return mae(df, models=["model"])["model"].mean()
```

**Reply by ArnodeVries** [reply]
Thanks, that worked.
